tests/mechanic/launcher_test.py

```diff
@@ -93,6 +99,116 @@ def info(self, *args, **kwargs):
         return self._info
 
 
+logging.basicConfig(level=logging.DEBUG)
+HOME_DIR = os.path.expanduser("~")
+
+
+def create_config():
+    cfg = config.Config()
+    # collect some mandatory config here
+    cfg.add(config.Scope.application, "node", "root.dir", HOME_DIR + "/.rally/benchmarks")
+    cfg.add(config.Scope.application, 'reporting', 'datastore.type', None)
+    cfg.add(config.Scope.application, 'track', 'params', None)
+    cfg.add(config.Scope.application, 'system', 'env.name', "unittest")
+    cfg.add(config.Scope.application, 'mechanic', 'keep.running', False)
+    cfg.add(config.Scope.application, 'mechanic', 'runtime.jdk', 12)
+    cfg.add(config.Scope.application, 'mechanic', 'telemetry.devices', [])
+    cfg.add(config.Scope.application, 'mechanic', 'telemetry.params', None)
+
+    return cfg
+
+
+def create_metrics_store(cfg, car):
+    cls = metrics.metrics_store_class(cfg)
+    metrics_store = cls(cfg)
+    metrics_store.lap = 0
+
+    metrics_store.open(trial_id="test",
+                       track_name="test",
+                       trial_timestamp=datetime.datetime.now(),
+                       challenge_name="test",
+                       car_name=car.name)
+    return metrics_store
+
+
+def create_default_car():
+    return team.load_car(HOME_DIR + "/.rally/benchmarks/teams/default",
+                         ["defaults"],
+                         None)
+
+
+def create_provisioner(car, ver):
+    installer = provisioner.ElasticsearchInstaller(car=car,
+                                                   java_home=guess_java_home(),
+                                                   node_name="rally-node-0",
+                                                   node_root_dir=HOME_DIR + "/.rally/benchmarks/races/unittest",
+                                                   ip="0.0.0.0",
+                                                   all_node_ips=["0.0.0.0"],
+                                                   http_port=9200)
+    p = provisioner.BareProvisioner(cluster_settings={"indices.query.bool.max_clause_count": 50000},
+                                    es_installer=installer,
+                                    plugin_installers=[],
+                                    preserve=True,
+                                    distribution_version=ver)
+    return p
+
+
+class ProcessLauncherTests(TestCase):
```

Comment:
I think testing this gets much simpler once we only rely on the PID. Then we can just mock process startup and mock the check for a PID file. Once we expose a `provision` and `launch` subcommand this should also become way simpler to add this as an integration test in `integration-test.sh`. I'd imagine an integration test along those lines:

```
PROVISIONER_ID=$(esrally provision --distribution-version=6.8.0 --car=4gheap)
esrally start --provisioner-id=${PROVISIONER_ID}
# maybe do a check here that the node is actually reachable via HTTP
esrally stop --provisioner-id=${PROVISIONER_ID}
```